contracts/contracts/coordination/FlateRateFeeModel.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "./IFeeModel.sol";
+import "../../threshold/IAccessControlApplication.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/**
+* @title FlatRateFeeModel
+* @notice FlateRateFeeModel
+*/
+contract FlatRateFeeModel is IFeeModel {
+
+    IERC20 public immutable currency;
+    uint256 public immutable feeRatePerSecond;
+    IAccessControlApplication public immutable stakes;
+
+    constructor(IERC20 _currency, uint256 _feeRatePerSecond, address _stakes){
+        currency = _currency;
+        feeRatePerSecond = _feeRatePerSecond;
+        stakes = IAccessControlApplication(_stakes);
+    }
+
+    function getRitualInitiationCost(
```

Comment:
Since the 'paywall' here appears to be generating a public key for the first time, I'm wondering how a sponsor tops-up the availability horizon without a new ritual, and also if once they are conferred the`INITIATOR_ROLE`, they can do this unilaterally and whenever they want

Reply:
Good questions. Let's track these here https://github.com/nucypher/nucypher-contracts/issues/93